pkg/image/api/types.go

```diff
@@ -60,6 +65,23 @@ type ImageLayer struct {
 	Size int64
 }
 
+// ImageSignatureType represents a type of image's signature.
+type ImageSignatureType string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type ImageSignatureType
```

Comment:
We've decided that we prefer enumerating different types as points to different structs so something like:

```go
type ImageSignature struct{
    AtomicSignature *AtomicSignature
    GenericSignature *GenericSignature
}
```

 and only one pointer is allowed to be non-nil